### PR TITLE
LITE-27895 Add create ppr functionality

### PR DESCRIPTION
**Pull request by jonatrios**
Changes on this PR:

- Add create new PPR functionality.
- Refactor code.

### CODE REVIEW

**Review by Hairash — APPROVED** [review]
For me it's ok to merge this code now to be able to rebase on it and to demonstrate the feature.
But I think it might be better to refactor it later. Below I will write my minds about what can be improved.

File: connect_ext_ppr/models/configuration.py [hunk 1]
```diff
@@ -12,20 +12,22 @@ class Configuration(Model):
 
     PREFIX = 'CFL'
 
+    STATE = ConfigurationStateChoices
```
Comment (Hairash) [inline connect_ext_ppr/models/configuration.py:15]: There is a lot of refactoring in several files. I think it is more convenient to make a separate commit for refactoring.

File: connect_ext_ppr/schemas.py [hunk 2]
```diff
@@ -68,7 +68,11 @@ class ConfigurationCreateSchema(NonNullSchema):
     file: FileSchema
 
 
-class ConfigurationSchema(NonNullSchema):
+class PPRCreateSchema(NonNullSchema):
```
Comment (Hairash) [inline connect_ext_ppr/schemas.py:71]: Description could be provided by user so it is required here. But I have it in my PR, so I will add it after merge.

File: connect_ext_ppr/service.py [hunk 3]
```diff
@@ -90,3 +97,136 @@ def update_product(data, config, logger):
...
+        .order_by(desc(PPRVersion.version))
+        .with_for_update()
+        .limit(1)
+        .scalar()
+    ) or 0
+    new_version = version + 1
+    return new_version
+
+
+def create_ppr(ppr, context, deployment, db, client, logger):
```
Comment (Hairash) [inline connect_ext_ppr/service.py:115]: This function is very big and has a lot of logic. It might be more convenient to refactor it and to split to several separate functions.

File: connect_ext_ppr/service.py [hunk 4]
```diff
@@ -90,3 +97,136 @@ def update_product(data, config, logger):
...
+    else:
+        summary = {}
+        data = get_ppr_from_media(client, deployment.account_id, deployment.id, file_data.id)
+        dict_file = workbook_to_dict(pd.ExcelFile(data))
+        errors = validate_ppr_schema(dict_file) or []
+        if errors:
+            summary.update({'errors': errors})
+            status = PPRVersion.STATUS.failed
+    try:
+        file_instance = File(
```
Comment (Hairash) [inline connect_ext_ppr/service.py:203]: I propose to leave this logic related to models in webapp.py for uniformity

File: connect_ext_ppr/service.py [hunk 5]
```diff
@@ -90,3 +97,136 @@ def update_product(data, config, logger):
...
+    new_version = get_ppr_new_version(db, deployment)
+    config_kwargs = {}
+    status = PPRVersion.STATUS.ready
+    if not file_data:
+        active_configuration = (
+            db.query(Configuration)
+            .filter_by(
+                deployment=deployment.id,
+                state=Configuration.STATE.active,
+            ).first()
```
Comment (Hairash) [inline connect_ext_ppr/service.py:126]: Better to use .one_or_none(), I suppose, to catch situation if there are more then one active configuration.

File: connect_ext_ppr/service.py [hunk 6]
```diff
@@ -90,3 +97,136 @@ def update_product(data, config, logger):
...
+                client, deployment.account_id, deployment.id, previous_ppr.file,
+            )
+        else:
+            logger.info(
+                f"Start creation of PPR version {product_info}"
+                f" based on previous product information.",
+            )
+        items = list(get_product_items(client, deployment.product.id))
+
+        file, writer, wb = get_base_workbook(data)
```
Comment (Hairash) [inline connect_ext_ppr/service.py:166]: All this file operations might be moved to the separate function

File: connect_ext_ppr/static/index.html [hunk 7]
```diff
@@ -11,7 +11,7 @@
     <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
     <link href="https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,400;0,500;0,700;1,400;1,500;1,700&display=swap" rel="stylesheet">
     <title>Index</title>
-<script defer src="vendors.c37f1993399243c8d2a4.js"></script><script defer src="index.f0239835f73a360467d7.js"></script><link href="index.5c5477c3e81eb95e1456.css" rel="stylesheet"></head>
+<script defer src="vendors.0b82e61f94dfa88a3bd2.js"></script><script defer src="index.acad834064fd73a90d98.js"></script><link href="index.5c5477c3e81eb95e1456.css" rel="stylesheet"></head>
```
Comment (Hairash) [inline connect_ext_ppr/static/index.html:14]: Why do these changes occur? I have the same in my PRs. Maybe we need to add static folder to .gitignore?

**Review by Hairash — APPROVED** [review]

File: connect_ext_ppr/service.py (hunk 4 above)
Comment (jonatrios) [reply connect_ext_ppr/service.py:203]: I agree on reduce the create function because is really big, but all related to file management, creating object is more related to the service layer than the webapp.

File: connect_ext_ppr/service.py [hunk 8]
```diff
@@ -90,3 +97,136 @@ def update_product(data, config, logger):
...
+        )
+        data = None
+        config_json = get_configuration_from_media(
+            client, deployment.account_id, deployment.id, active_configuration.file,
+        )
+        product_info = (
+            f"(product_id={deployment.product_id}, "
+            f"product_version={deployment.product.version})"
+        )
+        print(previous_ppr)
```
Comment (d3rky) [inline connect_ext_ppr/service.py:150]: print?

File: connect_ext_ppr/service.py (hunk 3 above)
Comment (d3rky) [reply connect_ext_ppr/service.py:115]: I agree here, at least there are two branches that can be done as separate functions --> uploading and generation

**Review by d3rky — CHANGES_REQUESTED** [review]

**Review by d3rky — APPROVED** [review]
